cromwell.example.backends/LocalExample.conf

```diff
@@ -0,0 +1,117 @@
+# This is an example of how you can use the LocalExample backend to define
+# a new backend provider. *This is not a complete configuration file!* The
+# content here should be copy pasted into the backend -> providers section
```

Comment:
@kshakir may need to correct me, but you may be able to add a static `include` to the top of these files to get the defaults automatically.

FWIW you *always* get the `reference.conf` automatically, so even with just this you should have a working Cromwell system

Reply:
Oh interesting! Can you give an example of how include would work? I'll add a note about the reference.conf.

Reply:
The idea (or what I was going for) isn't to include these separate files, but just to cleanly set out examples for the user to choose (and then copy paste into their final configuration).

Reply:
> you may be able to add a static include to the top of these files to get the defaults automatically

Yep! Stanzas are nice for copy-composing configs. For example grab different stanzas to build a conf that includes SLURM + mysql + call-caching. However for a "complete configuration file" technically all one needs is to prepend `include required(classpath("application.conf"))` above the `backend {}` stanza.

For example, a fully testable SLURM conf (based on [the conf used in CI](https://github.com/broadinstitute/cromwell/blob/3c3a3f85ba1229738265b11c3573ccb538b719c2/src/ci/resources/slurm_application.conf#L47)):

```hocon
include required(classpath("application.conf"))

backend {
  default = "SlurmNoDocker"
  providers {
    SlurmNoDocker {
      actor-factory = "cromwell.backend.impl.sfs.config.ConfigBackendLifecycleActorFactory"
      config {
        submit = """sbatch -J ${job_name} -D ${cwd} --wrap "/bin/bash ${script}""""
        kill = "scancel ${job_id}"
        check-alive = "squeue -j ${job_id}"
        job-id-regex = "Submitted batch job (\\d+).*"
      }
    }
  }
}
```

Something like that could get someone started. If there isn't going to be the include at the top, I **do** like the comment stating that this is a part of a full config, and "not a complete configuration file!"

Reply:
This might be a personal preference, but I would advise people to try to minimize dependencies (if they want to share their configuration, for example) and try to put all the config information into one file. The include is a neat trick, but it overcomplicates it. This isn't like writing a script where you want to import / include to have a cleaner application, it's just setting up a configuration file that is actually more optimal to just have one file. Does that make sense? The above could, if desired, go into some "advanced usage" part of the docs as to not confuse the new user.

Reply:
It would also potentially lead to bugs if there are equivalently defined objects in both the main config and the included files (e.g., duplicates of backends -> providers). It's much simpler to say "here is an example, take what you want from it and put it into your configuration file"

Reply:
Makes sense. I'm fine leaving the existing comment instead of the include.